### PR TITLE
feat: support Recommendation CRD

### PR DESCRIPTION
**Pull request by JolantaSk**

### CODE REVIEW

**Review by januszp-castai — APPROVED** [review]

File: internal/services/controller/crd/types.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package crd
```
Comment (varnastadeus) [inline internal/services/controller/crd/types.go:1]: package name is very generic, we should name it `recommendation` as it holds recommendation types

Comment (JolantaSk) [reply internal/services/controller/crd/types.go:1]: I kept the generic name crd as we might add more custom types in the future, not sure how realistic is that 😄

Comment (JolantaSk) [reply internal/services/controller/crd/types.go:1]: @varnastadeus if I rename the package to `recommendation` intellisense starts beating me up for types `RecommendationSpec` and `RecommendationStatus` as they start with the package name 🙈. 
Maybe `autoscalingv1` might be more descriptive 🤔

Comment (JolantaSk) [reply internal/services/controller/crd/types.go:1]: `autoscalingv1` is taken by k8s already, keeping it `crd`.

File: go.mod [hunk 2]
```diff
@@ -18,13 +18,14 @@ require (
...
+	github.com/spf13/pflag v1.0.5
 	github.com/spf13/viper v1.12.0
 	github.com/stretchr/testify v1.8.4
 	go.uber.org/goleak v1.2.1
 	golang.org/x/net v0.23.0
-	golang.org/x/sync v0.4.0
+	golang.org/x/sync v0.6.0
 	k8s.io/api v0.28.9
-	k8s.io/apimachinery v0.28.9
+	k8s.io/apimachinery v0.30.1
```
Comment (saumas) [inline go.mod:28]: We should keep the k8s libs within the same version to avoid any unexpected incompatibility issues. 🙏

**Review by saumas — CHANGES_REQUESTED** [review]
Rollback the version change and it should be good.

File: go.mod (hunk 2 above)
Comment (JolantaSk) [reply go.mod]: restored the version to the original one 🙏

**Review by saumas — APPROVED** [review]

**Pull request merged**